grpc/oob/bmc/bmc.go

```diff
@@ -359,6 +368,64 @@ func (m Action) BMCReset(ctx context.Context, rType string) (err error) {
 	return nil
 }
 
+// DeactivateSOL functionality for machines.
+func (m Action) DeactivateSOL(ctx context.Context) (err error) {
+	tracer := otel.Tracer("pbnj")
+	ctx, span := tracer.Start(ctx, "client.DeactivateSOL")
+	defer span.End()
+
+	host, user, password, parseErr := m.ParseAuth(m.DeactivateSOLRequest.Authn)
+	if parseErr != nil {
+		return parseErr
+	}
+	span.SetAttributes(attribute.String("bmc.host", host), attribute.String("bmc.username", user))
+	m.SendStatusMessage("working on SOL session deactivation")
+
+	opts := []bmclib.Option{
+		bmclib.WithLogger(m.Log),
+		bmclib.WithPerProviderTimeout(common.BMCTimeoutFromCtx(ctx)),
+		bmclib.WithIpmitoolPort("623"),
+	}
+
+	client := bmclib.NewClient(host, user, password, opts...)
+
+	err = client.Open(ctx)
+	if err != nil {
+		span.SetStatus(codes.Error, "Permission Denied: "+err.Error())
+		return &repository.Error{
+			Code:    v1.Code_value["PERMISSION_DENIED"],
+			Message: err.Error(),
+		}
+	}
+
+	log := m.Log.WithValues("host", host, "user", user)
+	defer func() {
+		client.Close(ctx)
+		log.Info("closed connections", logMetadata(client.GetMetadata())...)
+	}()
+	log.Info("connected to BMC", logMetadata(client.GetMetadata())...)
+	m.SendStatusMessage("connected to BMC")
+
+	err = client.DeactivateSOL(ctx)
+	log = m.Log.WithValues(logMetadata(client.GetMetadata())...)
+	if err != nil {
+		span.SetStatus(codes.Error, "failed to deactivate SOL session: "+err.Error())
+		log.Error(err, "failed to deactivate SOL session")
+	}
+	if err != nil {
+		span.SetStatus(codes.Error, "failed to deactivate SOL session: "+err.Error())
+		m.SendStatusMessage(fmt.Sprintf("failed to deactivate SOL session"))
+		return &repository.Error{
+			Code:    v1.Code_value["UNKNOWN"],
+			Message: err.Error(),
+		}
+	}
```

Comment:
```suggestion
	if err != nil {
		span.SetStatus(codes.Error, "failed to deactivate SOL session: "+err.Error())
		log.Error(err, "failed to deactivate SOL session")
		return &repository.Error{
			Code:    v1.Code_value["UNKNOWN"],
			Message: err.Error(),
		}
	}
```

Reply:
Applied with the slight change of also keeping the `m.SendStatusMessage()` call, which seems to be done elsewhere...